modules/programs/git/default.nix

```diff
@@ -28,36 +65,14 @@ in
         description = "git user email";
       };
 
-      gpgSigningKey = mkOption {
-        type = with types; nullOr str;
+      signing = mkOption {
+        type = types.nullOr signModule;
```

Comment:
To me it's weird to allow a null here, but if that's what HM does, let's keep it like this

Reply:
Yea I was expecting an `enable` option, but they instead do it by checking for null.